server/binding.js

```diff
@@ -74,30 +76,35 @@ binding.query = {};
  *   xr: (number|undefined),
  *   numSamples: (number|undefined)
  * }} query
- * @param {string} bindingPath
- * @return {binding.Histogram|binding.Error}
+ * @param {string} dataPath
+ * @param {function(*)} callback
  */
-binding.query.histogram = function(query, bindingPath) {
+binding.query.histogram = function(db, query, dataPath, callback) {
   if (query.fileName === undefined) {
-    return {error: 'fileName is undefined'};
+    callback({error: 'fileName is undefined'});
   }
   if (query.chr === undefined) {
-    return {error: 'chr is undefined'};
+    callback({error: 'chr is undefined'});
   }
   var fileName = query.fileName;
   var chr = query.chr;
+  var bindingPath = dataPath + 'anonymous/' + binding.BINDING_PREFIX_;
   var file = bindingPath + fileName + '_chr/' + fileName + '_chr' + chr +
     '.bcwig';
-  var descriptionPath = bindingPath + fileName + '.desc';
   if (!fs.existsSync(file)) {
     var error = 'binding file ' + fileName + ' not found.';
     log.serverLog(error);
-    return {error: error};
+    callback({error: error});
   }
   var data = binding.getBinding_(file, query.xl, query.xr, query.numSamples);
-  data.gene = binding.getGene_(descriptionPath);
   data.chr = chr;
-  return data;
+  binding.getGene_(db, fileName, function(gene) {
+    if (gene.error) {
+      callback(gene);
+    }
+    data.gene = /** @type {string} */(gene);
+    callback(data);
```

Comment:
I mean do not define the callback to be function(*), try to list the attributes like function({...}).